### PR TITLE
Config the log module

### PR DESCRIPTION
**Pull request by Kayryu**
Can specify the way to output logs. (file of stdout) #281 

rely  cryptape/cita-common#221

### CODE REVIEW

File: cita-jsonrpc/src/main.rs [hunk 1]
```diff
@@ -124,18 +124,20 @@ use ws_handler::WsFactory;
...
-
     // todo load config
     let matches = App::new("JsonRpc")
         .version(get_build_info_str(true))
         .long_version(get_build_info_str(false))
         .author("Cryptape")
         .about("CITA JSON-RPC by Rust")
-        .args_from_usage("-c, --config=[FILE] 'Sets a custom config file'")
+        .args_from_usage("-c, --config=[FILE] 'Sets a custom config file'
+                                -s, --stdout 'Log to console'")
```
Comment (ouwenkg) [inline cita-jsonrpc/src/main.rs]: Something missing?

File: cita-auth/src/main.rs [hunk 2]
```diff
@@ -142,17 +142,20 @@ fn profiler(flag_prof_start: u64, flag_prof_duration: u64) {
...
-    info!("Version: {}", get_build_info_str(true));
-
     // init app
     let matches = App::new("auth")
         .version(get_build_info_str(true))
         .long_version(get_build_info_str(false))
         .author("Cryptape")
         .about("CITA Block Chain Node powered by Rust")
         .args_from_usage("-c, --config=[FILE] 'Sets a custom config file'")
+        .args_from_usage("-s, --stdout 'Log to console'")
```
Comment (ouwenkg) [inline cita-auth/src/main.rs]:
```
 .args_from_usage("-c, --config=[FILE] 'Sets a custom config file'
                                -s, --stdout 'Log to console'")
```
```
.args_from_usage("-c, --config=[FILE] 'Sets a custom config file'")
.args_from_usage("-s, --stdout 'Log to console'")
```
If these two style both work well, my advice is choose one.

Comment (Kayryu) [reply cita-auth/src/main.rs]: ok, I will amend it.

**Comment by Kayryu** [comment]
@ouwenkg  Need help!  The issue must be compiled after https://github.com/cryptape/cita-bft/pull/99 , https://github.com/cryptape/cita-forever/pull/37.

**Comment by ouwenkg** [comment]
> @ouwenkg Need help! The issue must be compiled after [cryptape/cita-bft#99](https://github.com/cryptape/cita-bft/pull/99) , [cryptape/cita-forever#37](https://github.com/cryptape/cita-forever/pull/37).

I got you. We will review these pr as soon as possile.

**Comment by kaikai1024** [comment]
> @ouwenkg Need help! The issue must be compiled after [cryptape/cita-bft#99](https://github.com/cryptape/cita-bft/pull/99) , [cryptape/cita-forever#37](https://github.com/cryptape/cita-forever/pull/37).

done

**Comment by Kayryu** [comment]
> > @ouwenkg Need help! The issue must be compiled after [cryptape/cita-bft#99](https://github.com/cryptape/cita-bft/pull/99) , [cryptape/cita-forever#37](https://github.com/cryptape/cita-forever/pull/37).
> 
> done

thanks

**Comment by kaikai1024** [comment]
cita-common is not submodule.
You need to replace the latest cita-common commit in the Cargo.lock.

**Comment by Kayryu** [comment]
> cita-common is not submodule.
> You need to replace the latest cita-common commit in the Cargo.lock.

I just use `cargo update -p logger`  to update cita-common， I had't added it as submodule

**Comment by kaikai1024** [comment]
> > cita-common is not submodule.
> > You need to replace the latest cita-common commit in the Cargo.lock.
> 
> I just use `cargo update -p logger` to update cita-common， I had't added it as submodule

it's not like that to update common, you need to replace old commit with new commit.

**Comment by Kayryu** [comment]
> it's not like that to update common, you need to replace old commit with new commit.

Are there any better idea to to that, I need manual replace one by one?

**Comment by kaikai1024** [comment]
> Are there any better idea to to that

vim: `%s/old_commit/new_commit/g`

**Comment by Kayryu** [comment]
> > Are there any better idea to to that
> 
> vim: `%s/old_commit/new_commit/g`

thanks, it seems that should learn more about vim to improve work.

**Comment by kaikai1024** [comment]
> > > Are there any better idea to to that
> > 
> > 
> > vim: `%s/old_commit/new_commit/g`
> 
> thanks, it seems that should learn more about vim to improve work.

Just use IDE to replace it. vscode or atom...

**Comment by Kayryu** [comment]
> Just use IDE to replace it. vscode or atom

it's not hacker. vim looks more cool.
`cargo update -p logger`  seem work well, it can replace all last commit.

**Comment by kaikai1024** [comment]
> cargo update -p logger seem work well, it can replace all last commit.
cool

File: cita-executor/src/main.rs [hunk 3]
```diff
@@ -157,14 +157,20 @@ impl Options {
...
-        .arg_from_usage("-c, --config=[FILE] 'Sets a switch config file'")
+        .args_from_usage(
+            "-c, --config=[FILE] 'Sets a switch config file'
+                          -s, --stdout 'Log to console'",
+        )
         .get_matches();
+
+    let stdout = matches.is_present("stdout");
+    micro_service_init!("cita-executor", "CITA:executor", stdout);
+
```
Comment (ouwenkg) [inline cita-executor/src/main.rs]: why executor missing this log info?
`info!("Version: {}", get_build_info_str(true));`

Comment (ouwenkg) [reply cita-executor/src/main.rs]: I know. In the code block next.

Comment (Kayryu) [reply cita-executor/src/main.rs]: My fault， I add it.

Comment (Kayryu) [reply cita-executor/src/main.rs]: it existed at 176 line.

**Pull request merged**